### PR TITLE
Labhub restart

### LINKED ISSUE #75: Expose early bootstrap hook to aid development and debugging
Sometimes it is necessary to test new versions of jupyterlab extensions, while it is possible to install a new version into an environment it won't get activated until jupyterlab is restarted. This is problematic since restarting wipes out changes. Having a bash script hook exposed to a user would allow for modifying python environment before jupyterlab is launched.

This would allow for significantly faster iteration than what is currently possible.

This can be as simple as:

```bash
if [ -f "${SOME_HOOK_LOCATION}" ]; then
   source "${SOME_HOOK_LOCATION}"
fi
```

Where `SOME_HOOK_LOCATION` is some path inside `/home/jovyan/`. 
With this in place it would also make it much easier to debug/develop startup scripts.

### Error Recovery

It is important to ensure that even if the script fails in a bad way it does not prevent user from logging in and modifying it to fix the problem.

### PR DESCRIPTION
**Pull request by Kirill888**
Rather than adding launch hook allow restarts of the `jupyter-labhub`.

Run `jupyter-labhub` in a loop, if it exits successfully -> quit, else run it again and repeat.

Also tees stdout and stderr to a file (`/tmp/labhub.log`) to aid with debugging.

Closes #75

### CODE REVIEW

File: docker/assets/jupyter-labhub [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+# Wrapper around jupyter-labhub to enable restarting jupyter-labhub without
+# stopping docker image.
+#
+# This is needed to debug installation of server extensions without re-building
+# entire docker images.
+
+set -o pipefail
+
+APP_LOG="/tmp/labhub.log"
```
Comment (alexgleith) [inline docker/assets/jupyter-labhub:11]: We shouldn't be diverting logs to a file, I don't think. Better to do it the Docker way and keep them flowing to stdout, and reading them via Kubernetes. That means that logs flow into the main systems still.

Actually, are you just logging the "exited" and "failure" ?

Comment (Kirill888) [reply docker/assets/jupyter-labhub:11]: And how will I see them in the current deployment?

When debugging broken serverextensions I'm currently completely blind.

Comment (Kirill888) [reply docker/assets/jupyter-labhub:11]: by the way logs still flow to stdout, that's why I'm using `tee` I'm just trying to make them visible to the developer where developer is.

Comment (alexgleith) [reply docker/assets/jupyter-labhub:11]: Ah, cool. I didn't know tee did that.

Yeah, I know you want to see the logs without going to Kubernetes. This seems like a neat solution. Only issue is if it logs a lot and the disk fills up for users. But it's in tmp so it is pretty short-term.

**Pull request merged**